src/include/connection.h

```diff
@@ -300,6 +300,13 @@ typedef TAILQ_HEAD(__wt_backuphash, __wt_backup_target) WT_BACKUPHASH;
 
 extern const WT_NAME_FLAG __wt_stress_types[];
 
+/*
+ * Access the array of all sessions. This field uses the Slotted Array pattern to managed shared
+ * accesses, if you are looking to walk all sessions please consider using the existing session walk
+ * logic. FIXME-WT-10946 - Add link to Slotted Array docs.
+ */
+#define WT_CONN_SESSIONS_GET(conn) ((conn)->session_array.__array)
```

Comment:
Sorry to reopen the old convo but I think this is too subtle to prevent mis-use. e.g. we could have shouty capitals "DO NOT USE" above the macro, and name `__array` to `__array_do_not_use`

Reply:
We're sort of duplicating the __pindex arg on the page intl union. Hopefully `__` does clarify that it shouldn't get used. I suppose we could update the macro to make it more obvious, but i don't think the array name needs to change.

Maybe, "WT_CONN_SESSIONS_GET_UNSAFE".

Reply:
Oh I see. In that case, what about adding to the comment on `__array` as well? e.g. `Session reference. Do not use this field directly`

Reply:
Yeah that works.

Reply:
b91cd8c7f